src/public/target/sections/CancerBiomarkers/custom/FilterTable.js

```diff
@@ -150,79 +157,165 @@ const styles = theme => ({
   },
 });
 
-const getBiomarkerOptions = rows => {
-  return _.uniq(rows.map(row => row.biomarker)).map(row => ({
-    label: row,
-    value: row,
-  }));
-};
-
-const getDiseaseOptions = rows => {
-  return _.uniq(
-    rows.reduce((acc, row) => {
-      row.diseases.forEach(disease => {
-        acc.push(disease.name);
-      });
-      return acc;
-    }, [])
-  ).map(disease => ({ label: disease, value: disease }));
-};
+const BATCH_SIZE = 1000;
 
-const getDrugOptions = rows => {
-  return _.uniq(rows.map(row => row.drugName)).map(row => ({
-    label: row,
-    value: row,
-  }));
-};
-
-const getAssociationOptions = rows => {
-  return _.uniq(rows.map(row => row.associationType)).map(row => ({
-    label: row,
-    value: row,
-  }));
-};
+const COUNT_QUERY = gql`
+  query CancerBiomarkersCount($ensgId: String!) {
+    target(ensemblId: $ensgId) {
+      id
+      cancerBiomarkers {
+        count
+      }
+    }
+  }
+`;
+
+const PAGE_QUERY = gql`
+  query CancerBiomarkersSectionQuery($ensgId: String!, $page: Pagination!) {
+    target(ensemblId: $ensgId) {
+      id
+      cancerBiomarkers(page: $page) {
+        uniqueDrugs
+        uniqueDiseases
+        uniqueBiomarkers
+        count
+        rows {
+          id
+          associationType
+          drugName
+          evidenceLevel
+          sources {
+            link
+            name
+          }
+          pubmedIds
+          target {
+            approvedSymbol
+          }
+          disease {
+            name
+            id
+          }
+        }
+      }
+    }
+  }
+`;
```

Comment:
Perhaps we should extract this to the section query file.

Reply:
I agree, I also quite like to have the queries in external .gql files. 
I was following along the lines of another (adverse events?) widget and just couldn't get the external queries to work, so just included it in the .js file for now.